### ISSUE TITLE
TypeError: predict() got an unexpected keyword argument 'typ' raised during cross-validation

### ISSUE THREAD

**Issue by ahardjasa** [tgsmith61591/pyramid#29]
#### Description
`TypeError: predict() got an unexpected keyword argument 'typ'` raised when `seasonal=False`, CV is performed (i.e. `out_of_sample_size` is passed) and a `d` of `0` is selected by `auto_arima`. 
I think this is because an ARMA model doesn't take a `typ` argument, but [here](https://github.com/tgsmith61591/pyramid/blob/1e4801008edd0c3dfb6b67fb0ae3ed0b2b93af72/pyramid/arima/arima.py#L302) `typ='linear'` is being passed.
#### Steps/Code to Reproduce
```python
from pyramid.arima import auto_arima
import pandas
auto_arima(pandas.Series([3.0, 4.0, 1.0, 2.0, 1.0, 2.0, 0.0, 1.0, 
0.0, 0.0, 2.0, 0.0, 1.0, 1.0, 2.0, 2.0]), 
out_of_sample_size=3, seasonal=False)
```
#### Versions
Linux-4.15.0-29-generic-x86_64-with-debian-buster-sid
Python 3.5.5 (default, Jul 24 2018, 13:20:46) 
[GCC 7.3.0]
Pyramid 0.6.5
NumPy 1.15.0
SciPy 1.1.0
Scikit-Learn 0.19.2
Statsmodels 0.9.0

**Comment by tgsmith61591** [comment]
Thanks for filling! And good sleuthing. I'll take a look into this and the fix will be part of 0.7.0

**Comment by tgsmith61591** [comment]
This was technically handled in #28 since the new code doesn't use `typ='linear'` anymore:

```python
        if cv_samples is not None:
            # get the predictions (use self.predict, which calls forecast
            # from statsmodels internally)
            pred = self.predict(n_periods=cv, exogenous=cv_exog)
            self.oob_ = scoring(cv_samples, pred, **self.scoring_args)

            # If we compute out of sample scores, we have to now update the
            # observed time points so future forecasts originate from the end
            # of our y vec
            self.add_new_observations(cv_samples, cv_exog)
        else:
            self.oob_ = np.nan
```

But to be sure, I've included some new unit tests for cases where `d=0` and `out_of_sample > 0`. See, for instance, `pyramid/arima/tests/test_arima.test_oob_for_issue_29`:

```python
# Test Issue #29 (d=0, cv=True) -----------------------------------------------
def test_oob_for_issue_29():
    y = np.array([3.0, 4.0, 1.0, 2.0, 1.0, 2.0, 0.0, 1.0,
                  0.0, 0.0, 2.0, 0.0, 1.0, 1.0, 2.0, 2.0])

    xreg = np.random.RandomState(1).rand(y.shape[0], 3)

    # Try for cv on/off, various D levels, and various Xregs
    for d in (0, 1):
        for cv in (0, 3):
            for exog in (xreg, None):

                # surround with try/except so we can log the failing combo
                try:
                    model = ARIMA(order=(1, d, 1),
                                  out_of_sample_size=cv).fit(y, exogenous=exog)

                    # If exogenous is defined, we need to pass n_periods of
                    # exogenous rows to the predict function. Otherwise we'll
                    # just leave it at None
                    if exog is not None:
                        xr = exog[:3, :]
                    else:
                        xr = None

                    _, _ = model.predict(n_periods=3, return_conf_int=True,
                                         exogenous=xr)

                except Exception:
                    print("Failing combo: d=%i, cv=%i, exog=%r"
                          % (d, cv, exog))
                    raise
```

The changes that make this functional are going to be released with v0.7.0
